### PR TITLE
Added the code to check for palindrome#3001

### PR DESCRIPTION
**Pull request by cintamg**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
Yes

Hey @ankitaggarwal23 I have written the code.

### CODE REVIEW

**Comment by cintamg** [comment]
@ankitaggarwal23 can you help to find out the problem with this check.

**Comment by ankitaggarwal23** [comment]
> @ankitaggarwal23 can you help to find out the problem with this check.

You can ignore it, there is some glitch at deepsource's end.

**Comment by cintamg** [comment]
PR for issue #3001

**Comment by cintamg** [comment]
Palindrome_or_not.c

File: C/dp/Palindrome_or_not.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+Input : 123454321
+
+Output : Palindrome
+_____________________
+
+Test case 3:
+
+Input : 12345
+
+Output : Not palindrome
```
Comment (HarshCasper) [inline C/dp/Palindrome_or_not.c]: You did not even bother to end the comment body. Follow proper conventions here. Add a new line at the EOF here as well

File: C/dp/Palindrome_or_not.c [hunk 2]
```diff
@@ -0,0 +1,94 @@
+/*
+To find whether the given string is a palindrome or not.
+*/
```
Comment (HarshCasper) [inline C/dp/Palindrome_or_not.c]: Add a bit more description from a layman's point of view

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by cintamg** [comment]
Hey @HarshCasper, I have made all the necessary changes. Please review and merge it.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Make README updates as well and remove extra lines of code

**Comment by cintamg** [comment]
Hey @HarshCasper, I have made all the necessary changes. Please review and merge it.

**Comment by cintamg** [comment]
@ankitaggarwal23 do I need to make any more changes.

**Comment by cintamg** [comment]
Hey, @HarshCasper @ankitaggarwal23 @anushkrishnav @siddharth25pandey, please do review my pr
Thank you.

File: C/dp/Palindrome_or_not.c [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+Test case 2:
+Input : 123454321
+Output : Palindrome
+_____________________
+
+Test case 3:
+Input : 12345
+Output : Not palindrome
+*/
```
Comment (HarshCasper) [inline C/dp/Palindrome_or_not.c]: Add a new line at the EOF

File: C/dp/Palindrome_or_not.c [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+_____________________
+
+Test case 2:
+Input : 123454321
+Output : Palindrome
+_____________________
+
+Test case 3:
+Input : 12345
+Output : Not palindrome
```
Comment (HarshCasper) [inline C/dp/Palindrome_or_not.c]: Remove extra I/O samples. Add Time and Space Complexity

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by cintamg** [comment]
@HarshCasper I have made the changes. Please do review my pr.

**Comment by cintamg** [comment]
Hey, @HarshCasper @ankitaggarwal23 @anushkrishnav @siddharth25pandey , please do review my pr
Thank you.

**Comment by cintamg** [comment]
@HarshCasper I request you to review PR
Thank You

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**